Lib/inspect.py

```diff
@@ -506,10 +506,13 @@ def _is_wrapper(f):
             return hasattr(f, '__wrapped__') and not stop(f)
     f = func  # remember the original func for error reporting
     memo = {id(f)} # Memoise by id to tolerate non-hashable objects
```

Comment:
While we are here: we should switch to use a dictionary `{id(o): o}`. The same approach that `deepcopy` uses to ensure that objects stay alive long enough.